consumer.go

```diff
@@ -39,8 +40,9 @@ func New(streamName string, opts ...Option) (*Consumer, error) {
 		logger: &noopLogger{
 			logger: log.New(ioutil.Discard, "", log.LstdFlags),
 		},
-		scanInterval: 250 * time.Millisecond,
-		maxRecords:   10000,
+		scanInterval:    250 * time.Millisecond,
```

Comment:
curious if we re-think the `scanInterval` instead of introducing a new option. i.e. could we reduce it to zero to achieve a similar outcome. Or if it's not set then ignore the `scanInterval`

Reply:
There's still value in `scanInterval` though, since it would indicate the polling interval when we are current on the shard.